test.js

```diff
@@ -138,6 +154,74 @@ test('many operations', function (t) {
   })
 })
 
+test('keys and values should not be serialized', function (t) {
+  var DATA = []
+  var ITEMS = [
+    123,
+    'a string',
+    Buffer.from('w00t'),
+    { an: 'object' }
+  ]
+  ITEMS.forEach(function (k) {
+    ITEMS.forEach(function (v) {
+      DATA.push({ key: k, value: v })
+    })
+  })
+
+  function Db (m, fn) {
+    var db = {
+      open: function (options, cb) {
+        process.nextTick(cb)
+      }
+    }
+    var wrapper = function () {
+      wrapper.idx++
+      fn.apply(null, arguments)
+    }
+    wrapper.idx = -1
+    db[m] = wrapper
+    return new DeferredLevelDOWN(db)
+  }
+
+  function noop () {}
+
+  t.test('put', function (t) {
+    var ld = Db('put', function (key, value, cb) {
```

Comment:
You're essentially spying on the `put` call, correct? Can you simplify it to a pattern like:

```js
var calls = []
doTheThingThatMakesTheCalls()
t.same(calls, [{ key: .., value: ..}, { key: .., value: ..}]
```

Reply:
Correct. Sure!